api/proxy_util_test.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type StrA struct {
+	StrB
+
+	Internal struct {
+		A int
+	}
+}
+
+type StrB struct {
+	Internal struct {
+		B int
+	}
+}
+
+func TestGetInternalStructs(t *testing.T) {
+	var proxy StrA
+
+	sts := GetInternalStructs(&proxy)
+	require.Len(t, sts, 2)
+
+	sa := sts[0].(*struct{ A int })
+	sa.A = 3
+	sb := sts[1].(*struct{ B int })
+	sb.B = 4
+
+	require.Equal(t, 3, proxy.Internal.A)
+	require.Equal(t, 4, proxy.StrB.Internal.B)
+}
```

Comment:
Something maybe a bit useful would be to have a test case with a nested Internal:
```
type StrC struct {
   Internal struct {
      Internal struct {}
   }
}
```

to codify that the inner Internals don't get picked up.